hooks/pre-exit

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```

Comment:
Hey @jtszalay! We are not strictly against the change, but this is the behavior in all our plugins and this particular change doesn't affect or is particularly related to this PR :) 
If you can change it back it will be good to merge (but we'll add it to the backlog to discuss and analyze)

We had a previous [discussion](https://github.com/buildkite-plugins/cache-buildkite-plugin/pull/57#issuecomment-1824754560) around this in case you want to check it out.

Thanks!

Reply:
Reverted! I had added the line because on the systems our runners are on (nixos) boxes we don't have bash at that path and rely on the /usr/bin/env to find it. Thanks for adding it to the backlog!

Reply:
Following up to say that we found a work around (after my branch was created) to use 
```
    services.envfs.enable = true; # Enabling this so that /bin/bash lookups use our nix bash.
```